front/lib/plans/subscription.ts

```diff
@@ -0,0 +1,247 @@
+import { Op } from "sequelize";
+
+import { front_sequelize, ModelId } from "@app/lib/databases";
+import { Plan, Subscription } from "@app/lib/models";
+import {
+  FREE_TEST_PLAN_DATA,
+  FREE_UPGRADED_PLAN_CODE,
+  PlanAttributes,
+} from "@app/lib/plans/free_plans";
+import { generateModelSId } from "@app/lib/utils";
+import logger from "@app/logger/logger";
+import { SubscribedPlanType } from "@app/types/user";
+
+export const getActiveWorkspacePlan = async ({
+  workspaceModelId,
+}: {
+  workspaceModelId: ModelId | null;
+}): Promise<SubscribedPlanType> => {
+  let activeSubscription: Subscription | null = null;
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
+
+  if (workspaceModelId) {
+    activeSubscription = await Subscription.findOne({
```

Comment:
Maybe we should findAll and raise an error (or log a warning) if we have > 1 ? Since the unicity is not really enforced at db level

Feel free to disregard ofc, this is nitpicking

Reply:
It's technically enforced at the db level with the hook post create.

We don't really want to scan all rows for nothing tho. Using findOne is equivalent to using limit: 1 in a findAll query, so I could decide to replace by a findAll with a limit let's say set to 2 and if I have more than one I log an error so we get a notif and we can fix the data.